src/pages/SupplierRequisitionPage.js

```diff
@@ -163,14 +179,22 @@ export class SupplierRequisitionPage extends React.Component {
    */
   refreshData = (newSearchTerm, newSortBy, newIsAscending) => {
     const { requisition } = this.props;
+    const { useThresholdMOS, orderType } = this.state;
 
-    this.updateDataFilters(newSearchTerm, newSortBy, newIsAscending);
-    const { searchTerm, sortBy, isAscending } = this.dataFilters;
+    if (newSearchTerm && newSortBy && newIsAscending) {
+      this.updateDataFilters(newSearchTerm, newSortBy, newIsAscending);
+    }
 
-    const data = requisition.items.filtered(
+    const { searchTerm, sortBy, isAscending } = this.dataFilters;
+    let data = requisition.items.filtered(
       'item.name BEGINSWITH[c] $0 OR item.code BEGINSWITH[c] $0',
       searchTerm
     );
+    if (useThresholdMOS) {
+      data = data.filter(requisitionItem =>
```

Comment:
i think this if statement should be the only change in this method ?

Reply:
done 9095693bb8516259926361a7bf6d4ca464bdad4f